src/copy.ml

```diff
@@ -1031,43 +1022,36 @@ let transferFileUsingExternalCopyprog
              rootFrom pathFrom rootTo fspathTo pathTo realPathTo
              update desc fp ress id useExistingTarget =
   Uutil.showProgress id Uutil.Filesize.zero "ext";
-  let prog =
+  let progWithArgs =
     if useExistingTarget then
       Prefs.read copyprogrest
     else
       Prefs.read copyprog
   in
-  let extraquotes = Prefs.read copyquoterem = `True
-                 || (  Prefs.read copyquoterem = `Default
-                    && Util.findsubstring "rsync" prog <> None) in
-  let addquotes root s =
-    match root with
-    | Common.Local, _ -> s
-    | Common.Remote _, _ -> if extraquotes then Uutil.quotes s else s in
   let fromSpec =
       (formatConnectionInfo rootFrom)
-    ^ (addquotes rootFrom
-         (Fspath.toString (Fspath.concat (snd rootFrom) pathFrom))) in
+    ^ (Fspath.toString (Fspath.concat (snd rootFrom) pathFrom)) in
   let toSpec =
       (formatConnectionInfo rootTo)
-    ^ (addquotes rootTo
-         (Fspath.toString (Fspath.concat fspathTo pathTo))) in
-  let cmd = prog ^ " "
-             ^ (Uutil.quotes fromSpec) ^ " "
-             ^ (Uutil.quotes toSpec) in
-  Trace.log (Printf.sprintf "%s\n" cmd);
+    ^ (Fspath.toString (Fspath.concat fspathTo pathTo)) in
+  Trace.log (progWithArgs ^ " " ^ fromSpec ^ " " ^ toSpec ^ "\n");
   Lwt_util.resize_region !copyprogReg (Prefs.read copymax);
-  Lwt_util.run_in_region !copyprogReg 1
-    (fun () -> External.runExternalProgram cmd) >>= fun (_, log) ->
-  debug (fun() ->
-           let l = Util.trimWhitespace log in
-           Util.msg "transferFileUsingExternalCopyprog %s: returned...\n%s%s"
-             (Path.toString pathFrom)
-             l (if l="" then "" else "\n"));
-  Uutil.showProgress id (Props.length desc) "ext";
-  finishExternalTransferOnRoot rootTo rootFrom
-    ((snd rootFrom, pathFrom, fspathTo, pathTo, realPathTo),
-     (update, desc, fp, ress, id))
+  let args = Str.split (Str.regexp "[ \t]+") progWithArgs in
+  match args with
+  | [] -> assert false
+  | prog :: _ ->
```

Comment:
I'm fine with what you have done here but I usually like to keep the diff small, unless the changes are a clear improvement. You don't have to change this but I would propose to use a one-liner that will keep the indentation of all following lines unchanged.
```suggestion
  let prog = match args with [] -> assert false | h :: _ -> h in
```
The same works with multi-line as well, of course.

Reply:
Thanks, I've made this change; I agree of course about small diffs.

Reply:
I have also updated `strings.ml`.